forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,258 @@
+        if not self.configs.digit_label:
+            # initialize class
+            self._index2class = self.configs.index2class
+            self._class2index = {v: k for k, v in self._index2class.items()}
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.set_up()
+        if "label" not in self.configs.forte_data_fields:
+            raise ProcessorConfigError(
+                "There must be a forte data field named 'label'"
+                " in the reader config."
+            )
+
+        if self.configs.text_fields is None:
+            self.configs.text_fields = [
+                df
+                for df in self.configs.forte_data_fields
+                if df is not None and df != "label"
+            ]
+        else:
+            if not set(self.configs.text_fields).issubset(
+                set(self.configs.forte_data_fields)
+            ):
+                raise ProcessorConfigError(
+                    "text_fields must be a subset of forte_data_fields."
+                    f"text_fields: {self.configs.text_fields}"
+                    f"forte_data_fields: {self.configs.forte_data_fields}"
+                    "Please correct text_fields and forte_data_fields in the"
+                    " configuration to satisfy the condition."
+                )
+
+    def _collect(  # type: ignore
+        self, csv_file: str
+    ) -> Iterator[Tuple[int, List[str]]]:
+        with open(csv_file, encoding="utf-8") as f:
+            data = csv.reader(f, delimiter=",", quoting=csv.QUOTE_ALL)
+            if self.configs.skip_k_starting_lines > 0:
+                for _ in range(self.configs.skip_k_starting_lines):
+                    next(data)
+            for line_id, line in enumerate(data):
+                yield line_id, line
+
+    def _cache_key_function(self, line_info: Tuple[int, List[str]]) -> str:
+        return str(line_info[0])
+
+    def _parse_pack(
+        self, line_info: Tuple[int, List[str]]
+    ) -> Iterator[DataPack]:
+        line_id, line = line_info
+        # content label title
+        pack = DataPack()
+
+        if len(self.configs.forte_data_fields) != len(line):
+            raise ProcessorConfigError(
+                "Forte data fields provided in config "
+                "doesn't have one-to-one correspondence with the actual data "
+                "fields from dataset.\n"
+                f"Data fields length: {len(self.configs.forte_data_fields)} \n"
+                f"Line length: {len(line)}"
+            )
+        df_dict = dict(zip(self.configs.forte_data_fields, line))
+
+        # it determines the order of concatenation
+        text_fields = self.configs.text_fields
+
+        # get text and ontology indices
+        text, input_ontology_indices = generate_text_n_input_ontology_indices(
+            text_fields, df_dict
+        )
+        pack.set_text(text)
+        # self.set_text(pack, text)
```

Comment:
Should use `self.set_text`, since `pack.set_text` will not call `replace_func` automatically, in fact, in the future we could even make `pack.set_text` to be private and always force people to use `self.set_text`